### PR TITLE
support building branched v2plugin at install time

### PR DESCRIPTION
**Pull request by chrisplo**
Adds a series of tasks in an include if v2plugin is to be built on the
target machine

install v2plugin only when not doing a local v2plugin build

netctl link for /usr/bin moved earlier as needed by startcontiv in the
v2plugin

Drive-by:
* contiv_v2plugin_image now defaults to contiv_network_version
* contiv_network_version default bumped to 1.1.7
* use file module to create directory

Signed-off-by: Chris Plock <chrisplo@cisco.com>

### CODE REVIEW

**Comment by unclejack** [comment]
@chrisplo: I don't like this approach. We shouldn't be producing any images or plugins in contiv/ansible. We should make the building of the v2plugin tarball part of the contiv/netplugin tarball building process.

**Review by unclejack — DISMISSED** [review]
It's not acceptable to build the v2plugin here. This should be part of contiv/netplugin. We can leverage something built (and published if applicable) via contiv/netplugin.

**Comment by chrisplo** [comment]
can you look again please @unclejack ?

File: roles/contiv_network/tasks/v2plugin.yml [hunk 1]
```diff
@@ -1,32 +1,46 @@
...
 
 - name: check if v2plugin has been installed on {{ run_as }} nodes
   shell: docker plugin ls | grep -n {{ contiv_v2plugin_image }}
   register: v2plugin_installed
+  changed_when: no
   ignore_errors: True
 
-- name: install v2plugin on {{ run_as }} nodes
+- name: check if v2plugin archive is in the contiv_cache
+  local_action: stat path=/var/contiv_cache/{{ contiv_v2plugin_tar_filename }} follow=yes
```
Comment (tiewei) [inline roles/contiv_network/tasks/v2plugin.yml:22]: why not use `stat` module and make it a local operation ?

File: roles/contiv_network/tasks/v2plugin_local_install.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+- name: Create v2plugin rootfs directory
+  file: state=directory path=/var/lib/v2plugin/rootfs
+
+- name: Unpack v2plugin from local archive
+  unarchive:
+    src: "/var/contiv_cache/{{ contiv_v2plugin_tar_filename }}"
+    dest: /var/lib/v2plugin/rootfs/
+
+- name: Copy config.json for plugin container
+  copy: src=/var/contiv_cache/config.json dest=/var/lib/v2plugin
```
Comment (tiewei) [inline roles/contiv_network/tasks/v2plugin_local_install.yml]: `rootfs` and `config.json` looks pretty ephemeral, can we make it in temp dir instead ?

File: roles/contiv_network/tasks/v2plugin.yml (hunk 1 above)
Comment (chrisplo) [reply roles/contiv_network/tasks/v2plugin.yml:22]: local_action is a local operation using the stat module, it's a shorthand for delegate_to: 127.0.0.1:
http://docs.ansible.com/ansible/latest/playbooks_delegation.html#delegation

File: roles/contiv_network/tasks/v2plugin_local_install.yml (hunk 2 above)
Comment (chrisplo) [reply roles/contiv_network/tasks/v2plugin_local_install.yml]: sure, will do that soon as the -1 gets cleared

**Comment by unclejack** [comment]
@chrisplo: Please go ahead and make the requested changes.

**Review by tiewei — APPROVED** [review]

**Comment by chrisplo** [comment]
tested with https://github.com/contiv/install/pull/307:
```
make CONTIV_INSTALLER_VERSION=devbuild CONTIV_ANSIBLE_COMMIT=build_v2plugin_on_demand CONTIV_ANSIBLE_OWNER=chrisplo NETPLUGIN_BRANCH=v2plugin_container_bash NETPLUGIN_OWNER=chrisplo build install-test-swarm-mode
```

**Comment by chrisplo** [comment]
@unclejack plz take a look with updates for final approval

**Review by unclejack — APPROVED** [review]
LGTM

**Pull request merged**